core/src/main/java/io/questdb/log/HttpLogAlertBuilder.java

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.log;
+
+import io.questdb.VisibleForTesting;
+import io.questdb.std.*;
+
+
+public class HttpLogAlertBuilder extends LogRecordSink {
+
+    static final String CRLF = "\r\n";
+    private static final String CL_MARKER = "#######";
+    private static final int CL_MARKER_LEN = CL_MARKER.length(); // number of digits available for contentLength
+    private static final int CL_MARKER_MAX_LEN = (int) Math.pow(10, CL_MARKER_LEN) - 1;
+    private static final int MARK_NOT_SET = -1;
+
+
+    private long mark = MARK_NOT_SET;
+    private boolean hasContentLengthMarker;
+    private long contentLengthEnd;
+    private long bodyStart;
+    private Sinkable footer;
+
+    public HttpLogAlertBuilder(LogAlertSocket alertSkt) {
+        this(alertSkt.getOutBufferPtr(), alertSkt.getOutBufferSize());
+    }
+
+    public HttpLogAlertBuilder(long address, long addressSize) {
+        super(address, addressSize);
+        contentLengthEnd = _wptr;
+        bodyStart = _wptr;
+    }
+
+    public HttpLogAlertBuilder putHeader(CharSequence localHostIp) {
+        clear();
+        put("POST /api/v1/alerts HTTP/1.1").put(CRLF)
+                .put("Host: ").put(localHostIp).put(CRLF)
+                .put("User-Agent: QuestDB/LogAlert").put(CRLF)
+                .put("Accept: */*").put(CRLF)
+                .put("Content-Type: application/json").put(CRLF)
+                .putContentLengthMarker();
+        put(CRLF); // header/body separator
+        bodyStart = _wptr;
+        return this;
+    }
+
+    public int $() {
+        if (footer != null) {
+            footer.toSink(this);
+        }
+
+        if (hasContentLengthMarker) {
+            // take the body length and format it into the ###### contentLength marker
+            int bodyLen = (int) (_wptr - bodyStart);
+            if (bodyLen > CL_MARKER_MAX_LEN) {
+                throw new LogError("Content too long");
```

Comment:
CL_MARKER_MAX_LEN  == 9999999 ->  9999999 / 1024 / 1024 -> 9Mb
Could we ever log a 9Mb message?.
It would stop the logging thread... what should I do?

Reply:
I have extended the length by two digits to support up to ~900MB and never have to check